bot/cogs/snakes.py

```diff
@@ -28,9 +42,32 @@ def __init__(self, bot: AutoShardedBot):
         :param name: Optional, the name of the snake to get information for - omit for a random snake
         :return: A dict containing information on a snake
         """
+        # TODO: Random will be done by fetching from Special:RandomInCategory/Venomous_snakes, or something
+        async with self.session.get(self.base_query.format(name)) as pg_resp, \
+                   self.session.get(self.secs_query.format(name)) as sc_resp, \
+                   self.session.get(self.img_query.format(name)) as img_resp:
+            data = await pg_resp.json()
+            secs = await sc_resp.json()
+            img = await img_resp.json()
+        tidbits = []
+        soup = bs4.BeautifulSoup(data['parse']['text']['*'])
+        for section in secs['parse']['sections']:
+            for tag in await self.aexec(None, soup.find(id=section['anchor']).find_all_next):  # FIXME: inefficient...?
+                if tag.name == 'p':
+                    try:
+                        tidbits.append(rBRACK.sub('', rSENTENCE.match(tag.text)[0]))
+                    except TypeError:
+                        pass
+                    break
+        try:
+            pgid = str(data['parse']['pageid'])
+            imglink = img['query']['pages'][pgid]['thumbnail']['source']
+        except KeyError:
+            imglink = ''
+        return {'image': imglink, 'tidbits': tidbits}
 
-    @command()
-    async def get(self, ctx: Context, name: str = None):
+    @commands.command()
+    async def get(self, ctx, name: str.title = None):
```

Comment:
You removed the type hinting for `ctx`. This will break things. Also, you can't type hint `str.title`.

Reply:
huh, what would the Context typehint do and/or break? (are you folks running a type checker?)

The str.title typehint is a converter -- see Rapptz/discord.py#1064 (bad pr on my part but it's there) and [the documentation on converters](http://rapptz.github.io/discord.py/docs/ext/commands/commands.html#converters) (so hinting to just str is redundant as far as ext.commands is concerned)

[BUT: I've also just realized I don't even *want* to convert with str.title(), because Wikipedia articles are titled in `This case`. Oops]